gql/builder.go

```diff
@@ -38,12 +39,20 @@ var graphqlKinds = map[reflect.Kind]graphql.Type{
 	reflect.String:  graphql.String,
 }
 
+type ListFunctions struct {
+	SortField  string
+	SortOrder  string
+	FilterJSON string
+}
+
 // QueryReporter defines the interface used to report details on the GraphQL queries being performed.
 // Implementations must be concurrency safe and added to the request context using QueryReporterContextKey as the
 // context value key.
 type QueryReporter interface {
 	// QueriedField is called by the default resolve function used for fields within the GraphQL query.
 	QueriedField(string) error
+	// QueriedListFunctions is called by the list resolve function used for filtered fields within the GraphQL query.
+	QueriedListFunctions(string, ListFunctions) error
```

Comment:
This is a breaking change. I think you should put QueriedListFunctions into a new interface, maybe QueryFunctionReporter